examples/src/bin/server_acceptor.rs

```diff
@@ -225,7 +225,8 @@ impl TestPki {
         // based on the ClientHello (e.g. selecting a different certificate, or customizing
         // supported algorithms/protocol versions).
         let mut server_config = ServerConfig::builder()
-            .with_safe_defaults()
+            .with_safe_default_protocol_versions()
+            .unwrap()
             .with_client_cert_verifier(verifier)
             .with_single_cert(
                 vec![self.server_cert_der.clone()],
```

Comment:
I think one downside of this branch is that the paved-road where you create a builder with the default crypto provider, and then use `with_safe_default_protocol_versions()` to use default protocol versions requires an `unwrap` before continuing the build.

In the case where the default provider is used we have confidence there won't be an error result: the default ciphersuites from *ring* and aws-lc-rs are compatible with all protocol versions. The potential for error only exists when `builder_with_provider()` is used.

Perhaps a more ergonomic solution is to have:

* `ServerConfig::builder()` -> builds a server config with the default crypto provider and the safe protocol defaults, transitions to `WantsVerifier` state without needing an unwrap.
* `ServerConfig::builder_with_protocol_versions()` -> builds a server config with the default crypto provider, and a specified set of protocol versions, transitions to `WantsVerifier` without needing an unwrap.
* `ServerConfig::builder_with_provider()` -> builds a server config with a specified provider, transitions to `WantsVersions` where both `with_safe_default_protocol_versions()` and `with_protocol_versions()` allow customization of protocol versions, but requires handling a `Result` to get to `WantsVerifier`

In general this relates to the "Multiple Unwraps" problem called out in https://github.com/rustls/rustls/issues/1372

Reply:
I implemented this approach in [572f22b](https://github.com/rustls/rustls/pull/1628/commits/572f22b7b433485eeb9cb0454857a709d7fa9bbb) and I think it's an improvement.

If preferred I could break that out into a separate PR. In the meantime I'll resolve this thread.